### ISSUE TITLE
EditContext Crash When Using Attributes::ChangeValidate with Spawnable

### ISSUE THREAD

**Issue by AMZN-Gene** [o3de/o3de#11239]
**Describe the bug**
The editor crashes when attempting to using the ChangeValidate attribute for reflecting Spawnables or PhysicsMaterial to the editor.

**Steps to reproduce**
1. Take any component and add an EditContext DataElement that reflects a Spawnable 
Example                 
 ```
editContext->Class<MyClass>("My Class", "")
  ->DataElement(AZ::Edit::UIHandlers::Default, &MyClass::m_spawnableAsset, "My Spawnable Asset", "")
   ->Attribute(AZ::Edit::Attributes::ChangeValidate, &NetworkSpawnable::ValidatePotentialSpawnableAsset)
```
Where _m_spawnableAsset_ is _AZ::Data::Asset<AzFramework::Spawnable>_
2. Add on attribute to validate the change 
3. Run the Editor, attach the test component to an entity, and try selecting a spawnable in the component property

**Expected behavior**
Editor doesn't crash

**Actual behavior**
Editor crashes due to dereferencing a nullptr

```
>	EditorLib.dll!AzToolsFramework::TypedPropertyHandler_Internal<AZ::Data::Asset<AZ::Data::AssetData>,AzToolsFramework::PropertyAssetCtrl>::WriteGUIValuesIntoTempProperty_Internal(QWidget * widget, void * tempValue, const AZ::Uuid & propertyType, AZ::SerializeContext * serializeContext) Line 516	C++
 	ComponentEntityEditorPlugin.dll!AzToolsFramework::ReflectedPropertyEditor::Impl::RequestWrite(QWidget * editorGUI) Line 1411	C++
 	[Inline Frame] EditorLib.dll!AZStd::Internal::INVOKE(void(AzToolsFramework::PropertyEditorGUIMessages::*)(QWidget *) &) Line 177	C++
 	[Inline Frame] EditorLib.dll!AZStd::invoke(void(AzToolsFramework::PropertyEditorGUIMessages::*)(QWidget *) &) Line 54	C++
 	[Inline Frame] EditorLib.dll!AZ::EBusEventProcessingPolicy::Call(void(AzToolsFramework::PropertyEditorGUIMessages::*)(QWidget *) &) Line 437	C++
 	EditorLib.dll!AZ::Internal::EBusContainer<AzToolsFramework::PropertyEditorGUIMessages,AzToolsFramework::PropertyEditorGUIMessages,0,1>::Dispatcher<AZ::EBus<AzToolsFramework::PropertyEditorGUIMessages,AzToolsFramework::PropertyEditorGUIMessages>>::Broadcast<void (__cdecl AzToolsFramework::PropertyEditorGUIMessages::*)(QWidget *),AzToolsFramework::PropertyAssetCtrl * const &>(void(AzToolsFramework::PropertyEditorGUIMessages::*)(QWidget *) && func, AzToolsFramework::PropertyAssetCtrl * const & <args_0>) Line 1360	C++
 	[Inline Frame] EditorLib.dll!MotionPropertyWidgetHandler::CreateGUI::__l2::<lambda_1>::operator()(AZ::Data::AssetId) Line 22	C++
 	[Inline Frame] EditorLib.dll!QtPrivate::FunctorCall<QtPrivate::IndexesList<0>,QtPrivate::List<AZ::Data::AssetId>,void,`MotionPropertyWidgetHandler::CreateGUI'::`2'::<lambda_1>>::call(MotionPropertyWidgetHandler::CreateGUI::__l2::<lambda_1> &) Line 146	C++
 	[Inline Frame] EditorLib.dll!QtPrivate::Functor<`MotionPropertyWidgetHandler::CreateGUI'::`2'::<lambda_1>,1>::call(MotionPropertyWidgetHandler::CreateGUI::__l2::<lambda_1> & f, void *) Line 256	C++
 	EditorLib.dll!QtPrivate::QFunctorSlotObject<`MotionPropertyWidgetHandler::CreateGUI'::`2'::<lambda_1>,1,QtPrivate::List<AZ::Data::AssetId>,void>::impl(int which, QtPrivate::QSlotObjectBase * this_, QObject * r, void * * a, bool * ret) Line 443	C++
 	[External Code]	
 	[Inline Frame] EditorLib.dll!AzToolsFramework::PropertyAssetCtrl::OnAssetIDChanged(AZ::Data::AssetId) Line 403	C++
 	EditorLib.dll!AzToolsFramework::PropertyAssetCtrl::SetSelectedAssetID(const AZ::Data::AssetId & newID) Line 887	C++
 	[Inline Frame] EditorLib.dll!AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator>::Storage::GetData() Line 1781	C++
 	[Inline Frame] EditorLib.dll!AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator>::{dtor}() Line 166	C++
 	EditorLib.dll!AzToolsFramework::PropertyAssetCtrl::PopupAssetPicker() Line 839	C++

```
**Screenshots/Video**
![image](https://user-images.githubusercontent.com/32776221/184265981-c66ac1c3-6284-4561-903f-9176a285befe.png)

**Found in Branch**
development latest

**Comment by AMZN-Gene** [comment]
Blocking #9513

**Comment by lmbr-pip** [comment]
@AMZN-daimini Any updates on this issue?

**Comment by AMZN-daimini** [comment]
I got a repro this evening, will work on it and post my findings asap.

**Comment by AMZN-Gene** [comment]
Fixed #12189
